File: app/src/main/java/io/github/sithengineer/motoqueiro/data/sync/SyncComponent.java
package io.github.sithengineer.motoqueiro.data.sync;

import android.content.AbstractThreadedSyncAdapter;
import dagger.Subcomponent;
import io.github.sithengineer.motoqueiro.scope.ActivityScope;

@ActivityScope @Subcomponent(modules = { SyncModule.class })
public interface SyncComponent {
  void inject(AbstractThreadedSyncAdapter syncAdapter);
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/data/sync/SyncComponent.kt
package io.github.sithengineer.motoqueiro.data.sync

import android.content.AbstractThreadedSyncAdapter
import dagger.Subcomponent
import io.github.sithengineer.motoqueiro.scope.RideScope

@RideScope
@Subcomponent(modules = [SyncModule::class])
interface SyncComponent {
  fun inject(syncAdapter: AbstractThreadedSyncAdapter)
}
